### ISSUE TITLE
No documentation for the project build

### ISSUE THREAD

**Issue by nomiro** [errata-ai/vale-server#82]
I am trying to get Vale Server running on Alpine Linux, which uses musl instead of glibc, so I am trying to compile the binaries from source.

However, at the moment, this repository does not document the project build. As a consequence, your only option is to download one of the pre-built release artifacts.

For help with build instructions, I used https://github.com/probonopd/linuxdeployqt/blob/master/BUILDING.md, which outlines the build steps for a similar project.

With this guide, I was able to get a partial build working with a little trial and error:

1. `wget wget https://github.com/errata-ai/vale-server/releases/download/v2.0.0/Vale-Server-2.0.0-linux.AppImage`
2. `chmod 755 Vale-Server-2.0.0-linux.AppImage`
3. `./Vale-Server-2.0.0-linux.AppImage --appimage-extract`
4. cd `squashfs-root`
3. Install the build dependencies (commands vary by platform)
4. `qmake-qt5`
5. `make`
   * This command eventually fails when it tries to copy a non-existent `build` directory from the current to the current directory
      * I couldn't wrap my head around what this was trying to do
   * As a temporary fix, I commented out these lines: https://github.com/errata-ai/vale-server/blob/master/vale-server.pro#L49-L58
   * Then, I re-ran `qmake-qt5` and `make`
     * This no longer errors out at the end

This is as far as I got.

Note, if you try to run `qmake-qt5` on a clone of the repository source files, it will exit with an error. The only time this command runs successfully is when I run it against the extracted AppImage.

**Comment by ChrisChinchilla** [comment]
@nomiro I actually need to get a build working to fix the #80 issue (on macOS), so thanks for this, and when I figure it out myself, I will create documentation.

**Comment by jdkato** [comment]
There are a few different required steps:

```bash
# Build the static files for the dashboard:
$ cd dashboard && gulp css js nunjucks icon img && cd -
# Build the `valelib` executable:
$ cd lib && make && cd -
# Build the app; these steps can be done in Qt Creator too 
$ qmake
$ make
```

**Comment by ChrisChinchilla** [comment]
> ```shell
> cd dashboard && gulp css js nunjucks icon img && cd -
> ```

```
 cd dashboard && yarn install && gulp css js nunjucks icon img && cd -
 ```

**Comment by ChrisChinchilla** [comment]
@jdkato when running `make` get `make: *** No rule to make target 'lib/bin/valelib', needed by 'Vale Server.app/Contents/MacOS/valelib'.  Stop.`

**Comment by jdkato** [comment]
Does the executable at `lib/bin/valelib` exist? All that step does is copy it from that location to `Vale Server.app/Contents/MacOS/valelib`.

**Comment by ChrisChinchilla** [comment]
@jdkato Nope, _bin_ doesn't appear to be a folder…

<img width="1032" alt="Screen Shot 2022-02-10 at 11 48 39" src="https://user-images.githubusercontent.com/42080/153391736-8a4e55c1-a16a-493e-945e-f0be921fc8f8.png">

Once I get this working, I can help with a lot of issues here ;)

**Comment by ChrisChinchilla** [comment]
Yes, Sparkle is causing issues, I can get the build to progress by doing a lot of manual massaging, but it's brittle. @jdkato When you built in the first place, where did you install Sparkle.Framework to get it to show in your path?

I am anticipating similar issues with the other 2 platforms and need to figure out a way to manage these dependencies, but one thing at a time…

**Comment by ChrisChinchilla** [comment]
@jdkato It definitely needed it when I tried, but I'll retry… I guess I was trying to avoid QT creator, but now i've installed it for the next try. Though this won't help with CI builds. Once I actually understand the process I'll fathom if there's a way to build outside of QT creator… One thing at a time…

**Comment by ChrisChinchilla** [comment]
A bunch of fiddling (that I will document) and I got this working… Until i hit an issue as I am compiling on Apple Silicon 😩 . For that you need Qt 6.2+ and it seems this was built with 5.2. Seeing if I can just do a switch and configure to get to the next step…
